src/mono/mono/utils/mono-hwcap-arm.c

```diff
@@ -22,7 +22,7 @@
 
 #include "mono/utils/mono-hwcap.h"
 
-#if defined(HAVE_SYS_AUXV_H) && !defined(HOST_ANDROID)
+#if defined(HAVE_GETAUXVAL) && !defined(HOST_ANDROID)
```

Comment:
`#if HAVE_GETAUXVAL` since it is using cmakedefine01 instead of cmakedefine.

Reply:
ah thanks, I've changed it to `cmakedefine` since that is what we mostly use in mono.

Reply:
Then it will break src/native/minipal case. Please use cmakedefine01 since this is a shared code.

Reply:
Mono has its own config.h.in where we set this, there should be no affect to the minipal.

Reply:
btw. this looks wrong in the existing native/libs code since we use cmakedefine01 for it too: https://github.com/dotnet/runtime/blob/fa1f2834dace0a34c53f1e1039b5af640fe51576/src/native/libs/System.Native/pal_io.c#L214

https://github.com/dotnet/runtime/blob/fa1f2834dace0a34c53f1e1039b5af640fe51576/src/native/libs/Common/pal_config.h.in#L25

Reply:
Since I added that cmakedefine01 https://github.com/dotnet/runtime/commit/c88c88a5f07325a70322cfc056949e8d52e4a04f, naturally I know what mono is using. :)\

Your change is wrong and will suffer from same thing what I have fixed yesterday https://github.com/dotnet/runtime/pull/66874.

Reply:
> btw. this looks wrong in the existing native/libs code since we use cmakedefine01 for it too: 

Yes, we should fix that too.

Reply:
I can't see why this wouldn't work, would you mind explaining why you think it'd break (on my phone right now so can't play with it)?

Reply:
minipal/getexepath is used by all: coreclr, mono, libs and corehost. When `HAVE_GETAUXVAL` is detected on a system (which is a case with most modern linux systems), we wanted to use `HAVE_GETAUXVAL` (and avoid the fallback, procfs path, since procfs is usually not reliable).

So there are three ways we can define a macro in cmake's config file (.in):

```
// 1. if detected, defines HAVE_GETAUXVAL, else skips the definition
#cmakedefine HAVE_GETAUXVAL

// 2. if detected, defines HAVE_GETAUXVAL as 1, else defines HAVE_GETAUXVAL as 0
#cmakedefine01 HAVE_GETAUXVAL

// 3. if detected defines HAVE_GETAUXVAL as 1, else skips the definition
#define HAVE_GETAUXVAL 1
```

Since minipal (and almost everything under src/native) is shared code, we should use the same semantics everywhere. Otherwise, we get `Wundef` with `3.` from minipal that is using `#elif HAVE_GETAUXVAL`.

Reply:
Ok so it's more a consistency thing rather than something that wouldn't work, got it. I'll change it.